src/components/PollViewer/PollViewer.vue

```diff
@@ -0,0 +1,425 @@
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<NcModal v-if="id"
+		size="small"
+		:container="container"
+		@close="dismissModal">
+		<div v-if="poll" class="poll-modal">
+			<div class="poll-modal__header">
+				<PollIcon :size="20" />
+				<span role="heading" aria-level="2">
+					{{ name }}
+				</span>
+			</div>
+			<p class="poll-modal__summary">
+				{{ pollSummaryText }}
+			</p>
+
+			<!-- options -->
+			<div v-if="modalPage === 'voting'" class="poll-modal__options">
+				<NcCheckboxRadioSwitch v-for="(option, index) in poll.options"
+					:key="'option-' + index"
+					:checked.sync="checked"
+					:value="index.toString()"
+					:type="isMultipleAnswers ? 'checkbox' : 'radio'"
+					name="answerType">
+					{{ option }}
+				</NcCheckboxRadioSwitch>
+			</div>
+
+			<!-- results -->
+			<div v-else-if="modalPage === 'results'" class="results__options">
+				<div v-for="(option, index) in poll.options"
+					:key="index"
+					class="results__option">
+					<div class="results__option-title">
+						<p>{{ option }}</p>
+						<p class="percentage">
+							{{ getVotePercentage(index) + '%' }}
+						</p>
+					</div>
+					<div v-if="getFilteredDetails(index).length > 0 || selfHasVotedOption(index)"
+						class="results__option__details">
+						<PollVotersDetails v-if="poll.details"
+							:container="container"
+							:details="getFilteredDetails(index)" />
+						<p v-if="selfHasVotedOption(index)" class="results__option-subtitle">
+							{{ t('spreed', 'You voted for this option') }}
+						</p>
+					</div>
+					<NcProgressBar class="results__option-progress"
+						:value="getVotePercentage(index)"
+						size="medium" />
+				</div>
+			</div>
+
+			<div v-if="isPollOpen" class="poll-modal__actions">
+				<!-- Submit vote button-->
+				<NcButton v-if="modalPage === 'voting'"
+					type="primary"
+					:disabled="disabled"
+					@click="submitVote">
+					{{ t('spreed', 'Submit vote') }}
+				</NcButton>
+				<!-- Vote again-->
+				<NcButton v-else
+					type="secondary"
+					@click="modalPage = 'voting'">
+					{{ t('spreed', 'Change your vote') }}
+				</NcButton>
+				<!-- End poll button-->
+				<NcActions v-if="canEndPoll"
+					force-menu
+					:container="container">
+					<NcActionButton class="critical" @click="endPoll">
+						{{ t('spreed', 'End poll') }}
+						<template #icon>
+							<FileLock :size="20" />
+						</template>
+					</NcActionButton>
+				</NcActions>
+			</div>
+		</div>
+		<LoadingComponent v-else class="poll-modal__loading" />
```

Comment:
Use this instead https://nextcloud-vue-components.netlify.app/#/Components/NcLoadingIcon
We will be probably removing that soon as we now have it from upstream.